src/redux/slices/reservationsSlice.js

```diff
@@ -0,0 +1,85 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const initialState = {
+    loading: false,
+    reservations: [],
+    error: '',
+};
+
+const API_URL = 'http://127.0.0.1:3000/api/v1';
+
+export const fetchReservations = createAsyncThunk('reservations/fetchReservations', async (token) => {
+    // const token = localStorage.getItem('authToken');    
+    try{
+        const response = await axios.get(`${API_URL}/reservations`, {
+            headers: { 
+                'Content-Type': 'application/json', 
+                'Authorization': `Bearer ${token}`,
+            },
+        });
+        return response.data;
+    }
+    catch(error){
+        console.log(error);
+    }
+});
+
+export const createReservation = createAsyncThunk('reservations/createReservation', async (user_id, course_id, city, date) => {
+    const token = localStorage.getItem('authToken');    
+    try{
+        await fetch(`${API_URL}/reservations`, {
+            method: 'POST',
+            body: JSON.stringify(
+                    user_id,
+                    course_id,
+                    city,
+                    date,
+            ),
+            headers: { 
+                'Content-Type': 'application/json',
+                'Authorization': `Bearer ${token}`,
+            },
+          });
+
+        } 
+    catch (error) {
+            console.log(error);
+        }
+});
+
+// export const deleteReservation = createAsyncThunk('reservations/deleteReservation', async (id) => {
+//     try{
+//         await fetch(`${API_URL}/reservations/${id}`, {
+//             method: 'DELETE',
+//             headers: {
+//                 'Content-Type': 'application/json',
+//                 'Authorization': `Bearer ${token}`,
+//             },
+//         });
+//     }
+//     catch(error){
+//         console.log(error);
+//     }
+// });
```

Comment:
- [ ] Kindly, remove this old commented code in order to improve the repo's cleanliness.  👍

Reply:
Thanks @Meltrust for the review. Changes done! our code is cleanliness!